### ISSUE TITLE
validation of id

### ISSUE THREAD

**Issue by amarao** [rhysd/actionlint#182]
I've just got a reprimand from Github actions whilst actionlint was totally OK:

```
The identifier 'Set production tag (semantic)' is invalid. IDs may only contain alphanumeric characters, '_', and '-'. IDs must start with a letter or '_' and and must be less than 100 characters.
```
I believe Actionlint may validate steps ID  to match Github requirements.

Example of a bad workflow:
```
"on":
  push:

jobs:
  foo:
    steps:
      - id: Set production tag (semantic)
        run: echo hello
```
Sorry for the issue instead of a PR, Go is not my strongest.

**Comment by rhysd** [comment]
Thanks for your suggestion. It should be detected by actionlint.
